src/main/java/Deadline.java

```diff
@@ -0,0 +1,15 @@
+package main.java;
+
+public class Deadline extends Task {
+    private String dueDate;
+
+    public Deadline(String taskName, String dueDate) {
+        super(taskName);
+        this.dueDate = dueDate;
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + dueDate + ")";
```

Comment:
can consider separating `"[D]"` into a constant instead, likewise for `Event` and `Todo` classes :)